packages/validator/src/plugins/validation/2and3/semantic-validators/walker.js

```diff
@@ -6,20 +6,18 @@
 
 // Assertation 2:
 // In specific areas of a spec, allowed $ref values are restricted.
+// [Removed]
```

Comment:
@dpopp07 @pyrooka While looking through the code to identify built-in rules, I noticed that I didn't actually remove the logic for the "incorrect_ref_pattern" legacy rule, so updated the PR with those changes.